cscs-checks/apps/cp2k/cp2k_check.py

```diff
@@ -12,7 +11,11 @@ class Cp2kCheck(rfm.RunOnlyRegressionTest):
     def __init__(self):
         self.valid_prog_environs = ['builtin']
         self.modules = ['CP2K']
-        self.executable = 'cp2k.psmp'
+        if (self.current_system.name in ['eiger']):
+            self.executable = '--cpu-bind=cores cp2k.psmp'
```

Comment:
This is not the best way to add this. You should do it with pipeline hooks:

```python
@rfm.run_before('run')
def set_cpu_binding(self):
    self.job.launcher.options = ['--cpu-bind=cores']
```

Reply:
Thanks @vkarak, I have implemented the change that you have requested with the pipeline hook